src/codegen_internal.h

```diff
@@ -1,6 +1,6 @@
 // This file is a part of Julia. License is MIT: http://julialang.org/license
 
-#if defined(LLVM38) && !defined(LLVM37)
+#if JL_LLVM_VERSION >= 30800
```

Comment:
The condition is actually `USE_ORCJIT && JL_LLVM_VERSION <= 30800` (roughly 3.7 <= llvm_ver < 3.8). It is better to remove the declaration in `jitlayers.cpp` instead and keep this one with the condition fixed.